added endpoint for getting upazila data and fixed the endpoint for ge…

NirobNabil opened the pull request:
…tting ziladata

NirobNabil committed:
added endpoint for getting upazila data and fixed the endpoint for getting ziladata

azmainamin commented on src/server/routes/upazila/upazila.routes.js:
```diff
@@ -0,0 +1,16 @@
+const router = express.Router(); // eslint-disable-line new-cap
+
+router
+	.route("/")
```
Add a line comment following the convention: 
/** GET /api/upazila/ - Get info about all Upazilas **/

azmainamin commented on src/server/routes/upazila/upazila.routes.js:
```diff
@@ -0,0 +1,16 @@
+
+router
+  .route("/:UpazilaName")
+  /** GET /api/zila/:zilaName - Get Zila information */
```
Comment is wrong here.

azmainamin commented on src/server/routes/upazila/upazila.routes.js:
```diff
@@ -0,0 +1,16 @@
+	.get(UpazilaCtrl.getAllUpazilas);
+
+router
+  .route("/:UpazilaName")
```
Rename to :upazilaName . JS variable names start with lowercase unless they are CONSTANTS.

azmainamin commented on src/server/controllers/upazila.controller.js:
```diff
@@ -0,0 +1,47 @@
+ * @returns {} JS Object containing information about the Upazila
+ */
+function getUpazilaInfoByName(req, res) {
+  const UpazilaName = req.params.UpazilaName.toUpperCase();
```
Rename to upazilaName.

azmainamin commented on src/server/controllers/upazila.controller.js:
```diff
@@ -0,0 +1,47 @@
+    })
+    .catch();
+}
+
```
You also need to write tests for these controllers. For reference, check the zila.spec.js. Let me know if I can help!

azmainamin committed:
fixed zila test

Shanzid01 committed:
Queries addressed

- added requested comments
- changed variable name
- wrong a (maybe not perfect) test for upazila controller

NirobNabil committed:
addressed the requested changes

Shanzid01 committed:
Merge pull request #6 from TheRSquaredProject/shanzid-update

Queries addressed

NirobNabil committed:
addressed change requests

NirobNabil committed:
Merge remote-tracking branch 'origin/shanzid-update' into upazila_endpoint

# Conflicts:
#	src/server/controllers/upazila.controller.js
#	src/server/routes/upazila/upazila.routes.js
#	src/server/tests/upazila.spec.js
#	src/server/tests/zila.spec.js

NirobNabil committed:
Merge remote-tracking branch 'origin/upazila_endpoint' into upazila_endpoint

NirobNabil committed:
Merge remote-tracking branch 'origin/master' into upazila_endpoint

# Conflicts:
#	package.json
#	src/config/express.js
